storage/s3/s3_test.go

```diff
@@ -98,4 +155,17 @@ func TestS3(t *testing.T) {
 		assert.NoError(err)
 		assert.Equal("hello", string(b))
 	})
+
+	t.Run("should delete the file", func(t *testing.T) {
+		s, err := gilfoyle.NewStorage(storage.AmazonS3)
+		assert.NoError(err)
+
+		ctx := context.Background()
+
+		err = s.Delete(ctx, "world")
+		assert.NoError(err)
+	})
+	if err = pool.Purge(resource); err != nil {
+		log.Fatalf("Could not purge resource: %s", err)
```

Comment:
You should return error using the testing interface : ` t.Error(t, err)`

or `assert.NoError(t, err)`